server/portal/apps/workspace/api/views.py

```diff
@@ -226,19 +247,22 @@ def post(self, request, *args, **kwargs):
                     job_post['parameterSet']['envVariables'] = [license_var]
                 del job_post['licenseType']
 
-            # TODOv3 need to check if execution system needs keys (https://jira.tacc.utexas.edu/browse/TV3-94)
-            # Get or create application based on allocation and execution system
-            apps_mgr = UserApplicationsManager(request.user)
-            print(apps_mgr)  # TODOv3 testing workaround (to avoid flake8 error)
-            # app = apps_mgr.get_or_create_app(job_post['appId'], job_post['allocation'])
-
-            # TODOv3 need to check if execution system needs keys (https://jira.tacc.utexas.edu/browse/TV3-94)
-            # code: UserApplicationsManager get_or_create_app)
-            # if app.exec_sys:
-            #     return JsonResponse({"response": {"execSys": app.exec_sys.to_dict()}})
-
-            if 'parameterSet' not in job_post:
-                job_post['parameterSet'] = {}
+            # Test file listing on relevant systems to determine whether keys need to be pushed manually
+            for system_id in list(set([job_post['archiveSystemId'], job_post['execSystemId']])):
```

Comment:
I'm getting an error here for missing execSystemId in the job_post

```core_portal_django         |   File "/srv/www/portal/server/portal/apps/workspace/api/views.py", line 251, in post
core_portal_django         |     for system_id in list(set([job_post['archiveSystemId'], job_post['execSystemId']])):
core_portal_django         | KeyError: 'execSystemId'
```

we aren't submitting the exec system in the job post and relying on the app definition, right?

Reply:
Ah, good call. Think we should either set the execSystemId in the job post during submission on the client side, where the backend is prepped to handle an execSystemId, or remove the execSystemId check on the backend and have to add it back later when we accommodate an execSystemId being set on the frontend?

Reply:
I don't know which would be better.  I think:

> Think we should either set the execSystemId in the job post during submission on the client side

but tbh, i think you gotta a better feel on which would be the better path forward